doc/setup.md

```diff
@@ -0,0 +1,292 @@
+
+The following shows the protocol for setting up signer in a 'trustless' network which has n-singers and threshold t which t is less than n.
+
+We suppose (or recommend) that all signers can communicate with each other using the efficient and secure protocol providing PFS(perfect forward secrecy)[^1], such as SSL/TLS protocol or Noise Protocol[^2].
+
+[^1]: For more information about PFS, see the section 'D.5.1.7 Cryptoperiod and protection lifetime' in the 'IEEE Standard Specifications for Public-Key Cryptography' [IEEE Standard Specifications for Public-Key Cryptography](https://perso.telecom-paristech.fr/guilley/recherche/cryptoprocesseurs/ieee/00891000.pdf)
+[^2]: [The Noise Protocol Framework](http://www.noiseprotocol.org/noise.html)
+
+To support for setting up the Tapyrus Signer Network, we provide a command-line utility `tapyrus-setup`
+
+## Step 1. Choose Network ID
+
+Choose Network ID, which is a 32-bit integer, and share among all signers.
+
+## Step 2. Generate key pair
+
+Signer generates node key pairs using `tapyrus-setup createkey`.
+
+```
+tapyrus-setup createkey
+
+output <node_private_key> <node_public_key>
+```
+
+This generates the private/public key pair based on secp256k1 elliptic curve, where:
+
+- `node_private_key` is a private key with an extend WIF format[^3].
+- `node_public_key` is a public key corresponding to `node_private_key` with a hex format string.
+
+[^3]: [BIP-178: Version Extended WIF](https://github.com/bitcoin/bips/blob/master/bip-0178.mediawiki)
+
+Each signer shares their own public key `node_public_key` to other signers.
+And then, they sort received public keys.
+
+In the following steps, Signers are supposed to be sorted by public keys and indexed as Signer[i] (i = 1, 2, 3, ..., n)
+
+## Step 3. Generate node verifiable secret shares.
+
+Signer[i] generates "node" VSS(verifiable secret share)[^4] with `tapyrus-setup createvss`.
+
+```
+tapyrus-setup createvss -public_key=1:<node_public_key[1]> -public_key=2:<node_public_key[2]> ... -public_key=n:<node_public_key[n]> -private_key=<node_private_key[i]> -nonce=<networkid> -threshold=<t>
+
+output:
+    <node_public_key[1]>: <node_vss[i, 1]>,
+    <node_public_key[2]>: <node_vss[i, 2]>,
+    ...
+    <node_public_key[n]>: <node_vss[i, n]]>,
+```
+
+where:
+
+- `node_public_key[]` is a sorted array of public keys of signers and input args `j:node_public_key[j]` (j = 1, 2, ..., n) is a index and public key generated by Signer[j] in Step 2.
+- `private_key[i]` is a private key of Signer[i] generated in Step 2.
+- `networkid` is Network ID we choose in Step 1. we use it as a nonce of encryption.
```

Comment:
Do you have concrete algorithm using this nonce?

Reply:
Added calculating rule of the nonce 
d0c10dd27dc567f9d88a2c4c8cec32574cffe4ff